handlers/homu_status/__init__.py

```diff
@@ -5,45 +5,84 @@
 
 from eventhandler import EventHandler
 
+WPT_RESULT_ARROW = '\xe2\x96\xb6'
 
-def check_failure_log(api, bors_comment):
-    # bors_comment would be something like,
-    # ":broken_heart: Test failed - [linux2](http://build.servo.org/builders/linux2/builds/2627)"  # noqa
-    # ... from which we get the relevant build result url
-    url = iter(re.findall(r'.*\((.*)\)', str(bors_comment))).next()
-    if not url:
-        return
 
-    # Substitute and get the new url
-    # (e.g. http://build.servo.org/json/builders/linux2/builds/2627)
-    json_url = re.sub(r'(.*)(builders/.*)', r'\1json/\2', url)
-    json_stuff = api.get_page_content(json_url)
-    if not json_stuff:
-        return
+def parse_log(log):     # gets the names of the failed tests
+    test_names = []
+    l = log.split()
+    for i, thing in enumerate(l):
+        if thing == WPT_RESULT_ARROW:
+            # We can blindly index the results here, because we can rely
+            # on the WPT output. It has the unexpected result, expected result
+            # and the test path following that unicode character
+            name = l[i + 4].split('/')[-1]
+            test_names.append(name)
+    return test_names
 
-    build_stats = json.loads(json_stuff)
 
+def get_failed_url(json_obj):
+    if not json_obj:
+        return
+
+    build_stats = json.loads(json_obj)
     build_log = []
     for step in build_stats['steps']:
         if 'failed' in step['text']:
             build_log = step['logs']
             break
 
-    failed_url = None
     for (name, log_url) in build_log:
         if name == 'stdio':
             failed_url = log_url
-            break
+            return failed_url
+
+
+def get_known_intermittents(api, test_names):
```

Comment:
The danger with this method is that we'll end up making one API request per test that fails. For test runs where a lot of tests fail, this could cause us to go over the API limit.